mautic/mautic
# Theme improvements

### Fixes
1. Placeholder text defined in the config.php didn't make it to the builder.
2. There was no difference between html and text slot type. Both rendered with wysiwyg editor in the builder. This PR changes this behavior. Html type is still the content editable with the editor. Text type is now rendered as text input.
3. In the Mauve theme if you wanted to write something into the purple boxes, the text was white and the background as well. This PR styles builder editable fields has transparent background. It's readable now and it more looks like the actual result which Builder generates.

### Improvements
1. Before the PR, if the slot type was unknown, it ended up with PHP error. Added fallback so if the slot type is not known, it'll render as HTML type.
2. Added textarea slot type as addition to text type. Can be used by theme creator for slots where he needs multiple lines of HTML tags free text. Although HTML tags can be used there as well.

### Caveats
The slots now rendered in a text input will show the HTML markup (which shouldn't be there at the first place), at least `<p>some title text</p>` in the builder but it won't change the functionality.

### Testing
1. Modify the placeholder text for example in the Mauve theme:
https://github.com/mautic/mautic/blob/staging/themes/Mauve/config.php#L20
2. Create a page with Mauve theme.

The text in the slot's placeholder text will be still the same no matter what you put into the config file. After the PR the placeholder text will change to your text.

3. Try to edit various slots and make sure it saves the values correctly.
4. Look at the previously created pages if the changes are BC.
